config/7.1.0/securebanking/ig/scripts/groovy/AddDetachedSig.groovy

```diff
@@ -16,7 +16,10 @@ import org.forgerock.http.protocol.Status
  *
  */
 
-SCRIPT_NAME = "[AddDetachedSig] - "
+def fapiInteractionId = request.getHeaders().getFirst("x-fapi-interaction-id");
+if(fapiInteractionId == null) fapiInteractionId = "No x-fapi-interaction-id"
+SCRIPT_NAME = "[AddDetachedSig] (" + fapiInteractionId + ") - "
```

Comment:
I wonder if it would be good to add something like `%h x-fapi-interaction-id: %header{x-fapi-interaction-id}` in the logback.xml pattern for all lines logged in IG.

Would look something like: <pattern>%d{HH:mm:ss:SSS} | %-5level | %thread | %logger{20} | %h x-fapi-interaction-id: %header{x-fapi-interaction-id}|%message%n%xException</pattern>

Reply:
Looks like a good suggestion!

Reply:
Thinking about this, we are dependent on customers then configuring logback correctly, and I think we might end up using the forgeops project to deploy IG, and just use our repo to build the mounted IG config (routes, scripts, jars) etc. So this would end up being customer config dependent, or require modifying the base configuration of IG and building their own IG image.

So I wonder if this method is more reliable and is one thing less for customers to have to get right?

Reply:
Tried this, but didn't get far with it. Will create a new issue to look at this at a later date.